lib/hal/debug.c

```diff
@@ -27,9 +28,23 @@ static const unsigned char systemFont[] =
 #include "font_unscii_16.h"
 };
 
+static void synchronizeVideoMode()
+{
+	VIDEO_MODE vm = XVideoGetMode();
+	SCREEN_WIDTH = vm.width;
+	SCREEN_HEIGHT = vm.height;
+	SCREEN_BPP = vm.bpp;
+	SCREEN_FB = XVideoGetFB();
+}
+
+static void writebackBuffers()
+{
+	asm __volatile__("sfence");
```

Comment:
This code expects to use `XVideo`; so this should just be something like `XVideoFlush();`.